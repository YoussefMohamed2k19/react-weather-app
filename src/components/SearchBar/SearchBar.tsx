import { inject, observer } from "mobx-react";
import { useRef } from "react";

interface ISearchBar {
  store?: any;
}

const SearchBar = inject("store")(
  observer(({ store }: ISearchBar) => {
    const { changeCityName } = store;
    const inputField = useRef<HTMLInputElement | null>(null);
    return (
      <form
        onSubmit={(e) => {
          e.preventDefault();
          changeCityName(inputField.current?.value);
        }}
      >
        <input
          ref={inputField}
          type="text"
          className="form-control my-3"
          placeholder="search ......"
        />
      </form>
    );
  })
);

export default SearchBar;
