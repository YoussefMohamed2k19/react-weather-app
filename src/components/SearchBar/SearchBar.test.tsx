import { render, screen } from "@testing-library/react";
import SearchBar from "./SearchBar";

beforeEach(() => {
  jest.clearAllMocks();
});

test("renders SearchBar Component", () => {
  expect(SearchBar).toMatchSnapshot();
});
