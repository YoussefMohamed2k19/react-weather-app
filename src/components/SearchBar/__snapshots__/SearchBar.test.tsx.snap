// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SearchBar Component 1`] = `
Object {
  "$$typeof": Symbol(react.forward_ref),
  "isMobxInjector": true,
  "render": [Function],
  "wrappedComponent": Object {
    "$$typeof": Symbol(react.memo),
    "compare": null,
    "type": [Function],
  },
}
`;
